libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -26,7 +26,7 @@ int lua_servo_set_output_pwm(lua_State *state) {
         return luaL_error(state, "Servo function (%d) is not a scriptable output", servo_function);
     }
 
-    if (output_value > UINT16_MAX) {
+    if (output_value > (int)UINT16_MAX) {
```

Comment:
Why this explicit cast?

By the way, why the different use of `luaL_checkinteger` and `luaL_checknumber` above? And why not use `luaL_checkint`?

Reply:
Tried doing a PX4 build and the compilier freaked out about implicit promotion of `uint16_t` to integer.

`luaL_checkinteger` will throw an error if the provided number doesn't have an exact integer representation. So when we are trying to find the channel number to move that's fine as this should be exact. However the PWM value might be calculated by the user with floating point math and may not be exactly an integer value. Rather then ask the user to make sure they have an int in the script it's easier to ask for a `luaL_checknumber` which will return a `lua_Number` (`float` in our case) which we then cast to int. `luaL_checkint` is deprecated.